beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/BeekeeperPathSchedulerApiaryIntegrationTest.java

```diff
@@ -107,98 +116,213 @@ public static void init() throws IOException, SQLException {
     amazonSQS.createQueue(QUEUE);
 
     mySqlTestUtils = new MySqlTestUtils(jdbcUrl, username, password);
-
-    alterPartitionEvent = new String(
-        IOUtils.toByteArray(BeekeeperPathSchedulerApiaryIntegrationTest.class.getResource("/alter_partition.json")),
-        UTF_8);
   }
 
   @AfterAll
-  public static void teardown() throws SQLException {
+  static void teardown() throws SQLException {
     amazonSQS.shutdown();
     mySqlTestUtils.close();
     sqsContainer.stop();
     mySQLContainer.stop();
   }
 
   @BeforeEach
-  public void setup() throws SQLException {
+  void setup() throws SQLException {
     amazonSQS.purgeQueue(new PurgeQueueRequest(ContainerTestUtils.queueUrl(sqsContainer, QUEUE)));
     mySqlTestUtils.dropTable(PATH_TABLE);
     mySqlTestUtils.dropTable(FLYWAY_TABLE);
     executorService.execute(() -> BeekeeperPathSchedulerApiary.main(new String[] {}));
-    await().atMost(Duration.ONE_MINUTE)
-        .until(BeekeeperPathSchedulerApiary::isRunning);
+    await().atMost(Duration.ONE_MINUTE).until(BeekeeperPathSchedulerApiary::isRunning);
   }
 
   @AfterEach
-  public void stop() throws InterruptedException {
+  void stop() throws InterruptedException {
     BeekeeperPathSchedulerApiary.stop();
     executorService.awaitTermination(5, TimeUnit.SECONDS);
   }
 
   @Test
-  public void typicalSchedule() throws SQLException {
-    String path = "s3://path";
-    amazonSQS.sendMessage(sendMessageRequest(format(alterPartitionEvent, path)));
-    await().atMost(30, TimeUnit.SECONDS)
-        .until(() -> mySqlTestUtils.rowsInTable(PATH_TABLE) == 1);
-    List<EntityHousekeepingPath> paths = mySqlTestUtils.getPaths();
-    EntityHousekeepingPath savedPath = paths.get(0);
-    assertPathFields(savedPath);
-    assertThat(savedPath.getPath()).isEqualTo(path);
-    assertMetrics();
+  void noActionsRequested() throws IOException {
+    AlterPartitionSqsMessage alterPartitionSqsMessage = new AlterPartitionSqsMessage(
+        "s3://unreferencedTableLocation",
+        "s3://partitionLocation",
+        "s3://unreferencedPartitionLocation",
+        false, false);
+    amazonSQS.sendMessage(sendMessageRequest(alterPartitionSqsMessage.getFormattedString()));
+    CreateTableSqsMessage createTableSqsMessage = new CreateTableSqsMessage(
+        "s3://unreferencedTableLocation", false, true);
+    amazonSQS.sendMessage(sendMessageRequest(createTableSqsMessage.getFormattedString()));
+
+    await().atMost(TIMEOUT, TimeUnit.SECONDS).until(() -> mySqlTestUtils.unreferencedRowsInTable(PATH_TABLE) == 0);
```

Comment:
I think this will resolve to true almost immediately (maybe before even the reader has a chance to poll). I think we can remove this test and place faith in our unit tests for this one.

Reply:
Same for the next two, anything which checks that there are zero rows in the table really - as it's difficult to know for sure that the message has been processed + rejected _before_ doing this check. Open to ideas for solving this one though.

Reply:
Hmm - good point on this. I think a good way to test this would be to check is to do the await on when the SQS is empty. Once that's empty we have a reasonable assumption that beekeeper has popped & processed that message so we should be able to continue with the mysql check

Reply:
🤔 I implemented it with what I had mentioned above. This could still lead to some race conditions as we aren't dictating when beekeeper is actually going to run it's action.

The only way I can see us setting this up to be more consistent is if we move the beekeeper execute startup in the `beforeEach` block into the tests themselves after we setup our queue to be what we expect.

Thoughts on that approach?

Reply:
Hmm, yes I don't think this solves the problem either, as we can still have a situation where the message is read, and then the test finishes, before Beekeeper does anything. Even if we start Beekeeper after the message has been put on the queue, as soon as the message is read we will have the same issue.

Originally, as with all integration tests, these were set up to test the happy path scenarios and ensure the application wires up correctly. I don't think we need to be this specific, and we can rely on our unit tests to ensure these paths are covered. In which case, I think we can remove these three. What do you think?

Reply:
works for me! good to have that context on how integration tests were setup originally.
Figured I'd take a crack at it at the very least, but went ahead and removed those tests.